### PR TITLE
[enhancement](stats) Optimize stats pre-load logic

### PR DESCRIPTION
**Pull request by Kikyou1997**
1. Don't do pre-load until stats table gets ready
2. Don't put pre-loaded unknown stats to cache

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java [hunk 1]
```diff
@@ -194,6 +203,9 @@ public Optional<ColumnStatistic> join() {
                         return Optional.of(c);
                     }
                 };
+                if (c == ColumnStatistic.UNKNOWN) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java:206]: I think we should not save Unknown stats into statsTable.

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java [hunk 2]
```diff
@@ -146,6 +147,14 @@ public void preHeat() {
     private void doPreHeat() {
         List<ResultRow> recentStatsUpdatedCols = null;
         long retryTimes = 0;
+        while (!StatisticsUtil.statsTblAvailable()) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java:150]: if the statsTbl is not available for a long time, does doPreHeat block fe start up?

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java:150]: no

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java (hunk 1 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java:206]: why

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java:206]: There're too many reasons to cause it be unknown even it actually exists, I think we should passively handle unknown

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsCache.java:206]: Ok, I got your idea.

**Review by englefly — APPROVED** [review]

**Pull request merged**